### PR TITLE
Add E0665

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/libsyntax_ext/diagnostics.rs [hunk 1]
```diff
@@ -94,4 +94,38 @@ Considering that this would be a long explanation, we instead recommend you to
...
+```compile_fail,E0665
+#[derive(Default)]
+enum Food {
+    Sweet,
+    Salty,
+}
+```
+
+The `Default` cannot be derived on an enum for the simple reason that the
+compiler doesn't know which value to pick "by default" whereas it can for a
```
Comment (varkor) [inline src/libsyntax_ext/diagnostics.rs]: I'm not sure the quotes are necessary here (or below).

Comment (GuillaumeGomez) [reply src/libsyntax_ext/diagnostics.rs]: Sounded more funny. :D

Comment (frewsxcv) [reply src/libsyntax_ext/diagnostics.rs]: Agreed, I don't think the quotes add anything and I'd be in favor of removing them, but it's not that big of a deal. r=me either away

**Review by frewsxcv — APPROVED** [review]

**Pull request merged**